subiquity/models/tests/test_filesystem.py

```
@@ -78,3 +159,702 @@ def test_errors(self):
+        self.fsm.probe()
+        all_disks = self.fsm.all_disks()
+
+        # id: sda
+        for idx, disk in enumerate(all_disks):
+            disk.id = os.path.basename(disk.path)
+
+        # pick two disks
+        disk1 = all_disks[0]
+        disk2 = all_disks[1]
+
+        bootpart = self.fsm.add_partition(disk=disk1, size=(2 * MB),
+                                          flag='boot')
+        # id: sda1
+        bootpart.id = os.path.basename(bootpart.path)
+        bootfs = self.fsm.add_filesystem(bootpart, 'fat32')
+        # id: sda1-fat32
+        bootfs.id = "%s-%s" % (bootpart.id, bootfs.fstype)
+        boot_mount = self.fsm.add_mount(bootfs, '/boot/efi')
+        # id: sda1-boot-efi
+        boot_mount.id = "%s%s" % (bootpart.id,
+                                  boot_mount.path.replace('/', '-'))
+
+        rootpart = self.fsm.add_partition(disk=disk1, size=(1 * MB))
+        # id: sda2
+        rootpart.id = os.path.basename(rootpart.path)
+        rootfs = self.fsm.add_filesystem(rootpart, 'ext4')
+        # id: sda2-ext4
+        rootfs.id = "%s-%s" % (rootpart.id, rootfs.fstype)
+        root_mount = self.fsm.add_mount(rootfs, '/')
+        # id: sda2-rootfs
+        root_mount.id = "%s-rootfs" % (rootpart.id)
+
+        swappart = self.fsm.add_partition(disk=disk2, size=(2 * MB),
+                                          flag='swap')
+        # id: sdb1
+        swappart.id = os.path.basename(swappart.path)
+        swapfs = self.fsm.add_filesystem(swappart, 'swap')
+        # id: sdb1-swap
+        swapfs.id = "%s-%s" % (swappart.id, swapfs.fstype)
+        swap_mount = self.fsm.add_mount(swapfs, "")
+        # id: sdb1-mnt-swap
+        swap_mount.id = "%s-mnt-swap" % swappart.id
+
+        # make sure this is a valid config
+        self.assertTrue(self.fsm.can_install())
+        content = yaml.dump({'storage': {'config': self.fsm.render()}},
+                            default_flow_style=False, indent=4)
+        print(content)
```

You probably want to delete this.

---

We can; I find that we'll just add it back in when we try to see what's going on if the unittest fails.  The print is swallowed by the harness unless there is a failure.